pandas/_libs/hashtable_class_helper.pxi.in

```diff
@@ -8,7 +8,72 @@ WARNING: DO NOT edit .pxi FILE directly, .pxi is generated from .pxi.in
 {{py:
 
 # name
-cimported_types = ['float32',
+complex_types = ['complex64',
+                 'complex128']
+}}
+
+{{for name in complex_types}}
+cdef kh{{name}}_t to_kh{{name}}_t({{name}}_t val) nogil:
+    cdef kh{{name}}_t res
+    res.real = val.real
+    res.imag = val.imag
+    return res
+
+cdef {{name}}_t to_{{name}}(kh{{name}}_t val) nogil:
+    cdef {{name}}_t res
+    res.real = val.real
+    res.imag = val.imag
+    return res
+
+cdef bint is_nan_kh{{name}}_t(kh{{name}}_t val) nogil:
+    return val.real != val.real or val.imag != val.imag
+{{endfor}}
+
+
+{{py:
+
+# name
+float_types = ['float64_t',
+               'float32_t']
+}}
+
+{{for c_type in float_types}}
+
+cdef bint is_nan_{{c_type}}({{c_type}} val) nogil:
```

Comment:
we don't define this for int types (as a no-op)? would it make the logic simpler if we did this (then we can *always* call a parameterized is_nan ?

Reply:
Actually, we already do. I've reshuffled  the code, so it becomes more obvious.